--- lhash_ori.h	Tue Mar 23 20:54:23 2021
+++ lhash.h	Tue Mar 23 20:55:49 2021
@@ -217,22 +217,6 @@
  * this header file is included but the resulting object is not linked against
  * the Crypto library (openssl#6912).
  */
-# ifdef __SUNPRO_C
-#  pragma weak OPENSSL_LH_new
-#  pragma weak OPENSSL_LH_free
-#  pragma weak OPENSSL_LH_insert
-#  pragma weak OPENSSL_LH_delete
-#  pragma weak OPENSSL_LH_retrieve
-#  pragma weak OPENSSL_LH_error
-#  pragma weak OPENSSL_LH_num_items
-#  pragma weak OPENSSL_LH_node_stats_bio
-#  pragma weak OPENSSL_LH_node_usage_stats_bio
-#  pragma weak OPENSSL_LH_stats_bio
-#  pragma weak OPENSSL_LH_get_down_load
-#  pragma weak OPENSSL_LH_set_down_load
-#  pragma weak OPENSSL_LH_doall
-#  pragma weak OPENSSL_LH_doall_arg
-# endif /* __SUNPRO_C */
 
 #ifdef  __cplusplus
 }
