From 3f3354f7293034ac8790cae71d7a506f350eaa31 Mon Sep 17 00:00:00 2001
From: <luoxinping@huawei.com>
Date: Thu, 4 Mar 2021 11:29:41 +0800
Subject: [PATCH] [Huawei]Handling deadlocks
---
 lib/multi.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/lib/multi.c b/lib/multi.c
index 249e360..0a7c1bd 100644
--- a/lib/multi.c
+++ b/lib/multi.c
@@ -592,14 +592,12 @@ static CURLcode multi_done(struct Curl_easy *data,
 
   process_pending_handles(data->multi); /* connection / multiplex */
 
-  CONNCACHE_LOCK(data);
   Curl_detach_connnection(data);
   if(CONN_INUSE(conn)) {
     /* Stop if still used. */
     /* conn->data must not remain pointing to this transfer since it is going
        away! Find another to own it! */
     conn->data = conn->easyq.head->ptr;
-    CONNCACHE_UNLOCK(data);
     DEBUGF(infof(data, "Connection still in use %zu, "
                  "no more multi_done now!\n",
                  conn->easyq.size));
@@ -651,7 +649,6 @@ static CURLcode multi_done(struct Curl_easy *data,
     CURLcode res2;
     connclose(conn, "disconnecting");
     Curl_conncache_remove_conn(data, conn, FALSE);
-    CONNCACHE_UNLOCK(data);
     res2 = Curl_disconnect(data, conn, premature);
 
     /* If we had an error already, make sure we return that one. But
@@ -674,7 +671,6 @@ static CURLcode multi_done(struct Curl_easy *data,
               "Connection #%ld to host %s left intact",
               conn->connection_id, host);
     /* the connection is no longer in use by this transfer */
-    CONNCACHE_UNLOCK(data);
     if(Curl_conncache_return_conn(data, conn)) {
       /* remember the most recently used connection */
       data->state.lastconnect = conn;
-- 
2.1.4

